src/core/imageUtils.js

```diff
@@ -7,13 +7,33 @@ export function getAddonIconUrl(addon) {
     : fallbackIcon;
 }
 
-export const getPreviewImage = (addon, { index = 0, full = true } = {}) => {
-  const preview = addon.previews.length && addon.previews[index];
-  if (preview) {
-    const previewSize = full ? 'image_url' : 'thumbnail_url';
-    return preview[previewSize] && isAllowedOrigin(preview[previewSize])
-      ? preview[previewSize]
-      : null;
+export const getPreviewImage = (
+  addon,
+  { full = true, useStandardSize = true } = {},
+) => {
+  if (!addon.previews.length) {
+    return null;
   }
-  return null;
+
+  let imageIndex = 0;
+
+  if (useStandardSize) {
```

Comment:
I just realized that this option doesn't make sense with `full: false` because that will return `thumbnail_url`, not `image_url`. 

Just to avoid confusion, I would like to see this function throw an error if `full === false && useStandardSize === true`. I don't see anything using `full: false` so it shouldn't break anything. The error can mention "not supported yet" with a readable explanation.

Reply:
that's a good point! let me update